src/celutil/array_view.h

```diff
@@ -231,38 +235,31 @@ class array_view<void>
         m_size(vec.size() * sizeof(T))
     {}
 
-    /** @copydoc array_view::array_view((const std::array<T, N> &ary) */
-    template<typename T> constexpr array_view(std::vector<T> &&vec) noexcept :
-        m_ptr(vec.data()),
-        m_size(vec.size() * sizeof(T))
-    {}
-
     /**
      * Copy another view.
      */
-    template<typename T> constexpr array_view(const array_view<T> &ary) :
-        m_ptr(ary.data()),
-        m_size(ary.size())
-    {}
+    constexpr array_view(const array_view&) = default;
 
-    /** @copydoc array_view::array_view(const array_view<T> &ary) */
-    template<typename T> constexpr array_view(array_view<T> &&ary) :
```

Comment:
No resources to steal so have left out the move constructors, copying is fine here.